app/src/main/java/com/sakaimobile/development/sakaiclient20/repositories/ResourceRepository.java

```diff
@@ -0,0 +1,74 @@
+package com.sakaimobile.development.sakaiclient20.repositories;
+
+import com.sakaimobile.development.sakaiclient20.models.sakai.resources.ResourcesResponse;
+import com.sakaimobile.development.sakaiclient20.networking.services.ResourcesService;
+import com.sakaimobile.development.sakaiclient20.persistence.access.ResourceDao;
+import com.sakaimobile.development.sakaiclient20.persistence.entities.Resource;
+
+import java.util.List;
+
+import io.reactivex.Completable;
+import io.reactivex.Single;
+
+public class ResourceRepository {
+
+    private ResourceDao resourceDao;
+    private ResourcesService resourcesService;
+
+    public ResourceRepository(ResourceDao resourceDao, ResourcesService service) {
+        this.resourceDao = resourceDao;
+        this.resourcesService = service;
+    }
+
+    /**
+     * Gets the stored site resources from room database
+     * @param siteId siteId of resources
+     * @return list of resources, can be observed on
+     */
+    public Single<List<Resource>> getSiteResources(String siteId) {
+        return resourceDao
+                .getSiteResources(siteId)
+                .firstOrError();
+    }
+
+    /**
+     * refresh the site resources, sets the siteIDs and then persists them
+     * @param siteId siteId of resources to request
+     * @return whether or not the operation was successful
+     */
+    public Single<List<Resource>> refreshSiteResources(String siteId) {
+        return resourcesService
+                .getSiteResources(siteId)
+                .map(ResourcesResponse::getResources)
+                .map(resources -> setSiteIdOfResources(siteId, resources))
```

Comment:
Oh I see, this is where the siteId is set on the resources. Then maybe change the comment in the deserializer and add a comment here to describe _why_ it's done this way, because the deserializer gives the impression that siteId is never set.